File: examples/joystick/joystick.go
package main

import (
	"fmt"
	"github.com/hybridgroup/gobot"
	"github.com/hybridgroup/gobot/api"
)

func main() {
	gbot := gobot.NewGobot()
	api := api.NewAPI(gbot)
	api.Port = "8080"

	work := func() {}

	robot := gobot.NewRobot("joystick",
		[]gobot.Connection{},
		[]gobot.Device{},
		work,
	)

	robot.AddCommand("joystick_event", func(params map[string]interface{}) interface{} {
		value := params["name"].(string) + ": "
		fmt.Println(value, params["position"])
		return value
	})

	robot.AddCommand("button_event", func(params map[string]interface{}) interface{} {
		value := params["name"].(string) + ": " + params["action"].(string)
		fmt.Println(value)
		return value
	})

	gbot.AddRobot(robot)
	api.Start()
	gbot.Start()
}

File: examples/joystick/joystick.js
var Cylon = require('cylon');

Cylon.robot({
  name: 'joystick',
  commands: function() {
    return {
      button_event: function(name, action) {
        console.log(name + ": " + action);
      },
      joystick_event: function(name, position) {
        if (position ) {
          console.log(name + ": " + position.x + "," + position.y);
        }
      }
    };
  }
});

Cylon.api('http', {
  host: '0.0.0.0',
  port: '8080',
  ssl:  false
});

Cylon.start();
